deploy/deploy.go

```diff
@@ -134,6 +135,35 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) error {
 	if err != nil {
 		return err
 	}
+
+	log.Infof("Checking kubectl versions.")
```

Comment:
this block of code should have test coverage

ideally we should add a test for all of deployment.Deploy with fakeCluster/Ingress/CNI/Controller implementations

but for now we could just move this added block to its own function and write a test for that

Reply:
ack there needs to be a test deploy but since there isn't ... not really worth trying to address this in this pr